core/src/main/java/org/hibernate/ogm/jpa/impl/OgmEntityManagerFactory.java

```diff
@@ -39,22 +41,26 @@ public OgmEntityManagerFactory(EntityManagerFactory hibernateEmf) {
 
 	@Override
 	public EntityManager createEntityManager() {
-		return new OgmEntityManager( this, hibernateEmf.createEntityManager() );
+		return new OgmEntityManager( this, hibernateEmf.createEntityManager().unwrap( EntityManagerImpl.class ) );
```

Comment:
Unwrapping is what one should do, but I agree the unwrap method is fishy. Unwrapping should check against `EntityManagetImpl` explicitly. In fact it should probably only be possible to unwrap to `org.hibernate.jpa.spi.AbstractEntityManagerImpl